WordPressKit/Scanner+extensions.swift

```diff
@@ -3,13 +3,13 @@ import Foundation
 extension Scanner {
     func scanQuotesText() -> [String] {
         var scanned = [String]()
-        var quote: NSString?
+        var quote: String?
         let quoteString = "\""
         while self.isAtEnd == false {
-            scanUpTo(quoteString, into: nil)
-            scanString(quoteString, into: nil)
-            scanUpTo(quoteString, into: &quote)
-            scanUpTo(quoteString, into: nil)
+            _ = scanUpToString(quoteString)
+            _ = scanString(quoteString)
+            quote = scanUpToString(quoteString)
+            _ = scanUpToString(quoteString)
```

Comment:
This change looks good.

It's understandably out of scope for this PR, I wish there had been already a test to verify it 😞

Reply:
I agree. I've spent some time digging in and:
- Looks like this extension is broken (or I don't understand the logic behind it in my tests).
- WP-iOS has its own very similar extensions for [NSScanner](https://github.com/wordpress-mobile/WordPress-iOS/blob/a3eae1d75066b28c215180cafd12e4947c74e556/WordPress/Classes/Categories/NSScanner+Helpers.m) and [NSAttributedString](https://github.com/wordpress-mobile/WordPress-iOS/blob/a3eae1d75066b28c215180cafd12e4947c74e556/WordPress/Classes/Categories/NSAttributedString+Util.m).
- This `Scanner` extension used only in `NSMutableAttributedString` extension which is unused in library and declared [internal](https://github.com/wordpress-mobile/WordPressKit-iOS/blob/22a68b62053d9a268a4fd093b205b83e53b49bf6/WordPressKit/NSMutableAttributedString%2Bextensions.swift#L3).

I've decided to decouple this issue into separate branch (and remove duplicated code from WP there):
https://github.com/wordpress-mobile/WordPressKit-iOS/tree/update/quotes-attributes-extension

Reply:
Great digging! Keep me posted on in because I now I'm curious to see how the cleanup will evolve 🍿